userspace/engine/rule_reader.cpp

```diff
@@ -222,57 +284,81 @@ static void read_item(
 	}
 	else if(item["rule"].IsDefined())
 	{
-		rule_loader::rule_info v;
-		v.ctx = ctx;
-		bool append = false;
+		std::string name;
+
+		// Using tmp context until name is decoded
+		rule_loader::context tmp(item, "rule", "", parent);
+		decode_val(item, "rule", name, tmp);
+
+		rule_loader::context ctx(item, "rule", name, parent);
+		rule_loader::rule_info v(ctx);
+		v.name = name;
+
+		v.append = false;
```

Comment:
Sure, that's fine. I initially thought that rule_reader simply parsed the objects from yaml into a list, with a separate pass for handling append. But since this version handles append while parsing the yaml, we can keep things simpler and have append just be a local variable.